fix(deps): vcpkg port fails - thread_system find_package fails despite installation

## What

The `kcenon-logger-system` vcpkg port fails to configure because `find_package(thread_system)` does not find the installed thread_system package, causing a FetchContent fallback that fails in the vcpkg sandbox.

## Root Cause

The logger_system uses `unified_find_dependency(thread_system)` which internally:
1. Tries `find_package(thread_system)` — fails (reason unclear, possibly target name mismatch)
2. Falls back to `FetchContent_Populate` — fails because:
   - vcpkg sandbox restricts network access during build
   - CMake 4.x CMP0170 policy deprecates `FetchContent_Populate`

Even though thread_system IS installed as a vcpkg dependency, the `UnifiedDependencies.cmake` module cannot locate it.

## Build Error

```
CMake Error at cmake/UnifiedDependencies.cmake:426 (add_subdirectory):
  add_subdirectory given source
  ".../_deps/threadsystem-src"
  which is not an existing directory.
```

```
-- Logger System: thread_system not found, using standalone mode
-- Configuring incomplete, errors occurred!
```

## Suggested Fix

The portfile should ensure the CMake find path is set up correctly, or the `UnifiedDependencies.cmake` should properly detect vcpkg-installed packages. Possible approaches:

1. In portfile: pass `-DLOGGER_USE_THREAD_SYSTEM_VCPKG=ON` or similar to skip FetchContent
2. In `UnifiedDependencies.cmake`: check for `thread_system::ThreadSystem` target (the actual exported name) before attempting FetchContent

## Reproduction

```bash
cd monitoring_system
./scripts/validate_vcpkg_chain.sh --registry ../vcpkg-registry --clean
```

## Context

Found during full Tier 0-5 vcpkg build chain validation (kcenon/monitoring_system#511).
Related: kcenon/database_system#432 (similar target name mismatch issue)

--- a/cmake/UnifiedDependencies.cmake
+++ b/cmake/UnifiedDependencies.cmake
@@ -116,7 +116,7 @@ set(_UNIFIED_FETCH_NAME_network_system "NetworkSystem")
 
 # find_package names
 set(_UNIFIED_PACKAGE_NAME_common_system "common_system")
-set(_UNIFIED_PACKAGE_NAME_thread_system "ThreadSystem")
+set(_UNIFIED_PACKAGE_NAME_thread_system "thread_system")
 set(_UNIFIED_PACKAGE_NAME_logger_system "LoggerSystem")
 set(_UNIFIED_PACKAGE_NAME_monitoring_system "MonitoringSystem")
 set(_UNIFIED_PACKAGE_NAME_container_system "ContainerSystem")

